pkg/registries/topo.go

```diff
@@ -49,6 +49,12 @@ func (g *topoGraph) getNode(node topoNode) *internalTopoNode {
 	return res
 }
 
+// AddNode creates and adds a node to the graph if it does not already exist.
+// If the node already exists, this function does nothing.
+func (g *topoGraph) AddNode(node topoNode) {
+	g.getNode(node)
```

Comment:
Non-blocking: I’d weakly prefer an explicit `_ = g.getNode(node)` to draw attention to the side effect of the function.